code/modules/jobs/job_types/shaft_miner.dm

```diff
@@ -112,8 +112,8 @@
 			var/obj/item/stack/sheet/animalhide/goliath_hide/plating = new()
 			explorer_suit.hood.attackby(plating)
 	for(var/obj/item/gun/energy/recharge/kinetic_accelerator/accelerator in miner_contents)
-		var/obj/item/knife/combat/survival/knife = new(accelerator)
-		accelerator.bayonet = knife
+		var/datum/component/bayonet_attachable/bayonet = accelerator.GetComponent(/datum/component/bayonet_attachable)
+		bayonet.add_bayonet(new /obj/item/knife/combat/survival(accelerator))
```

Comment:
This is a little gross, even though I see we also do it below and it's probably fine because it's just for admin code outfit.
That said, and stop me if this is insane, couldn't you just do `attackby` passing the person wearing the outfit as the user? :clueless:

Reply:
...that feels like begging for trouble doesn't it